accepted/future-releases/extension-types/feature-specification.md

```diff
@@ -0,0 +1,1131 @@
+types (except bottom types) cannot be assigned to extension types without
+a cast. Similarly, null cannot be assigned to an extension type without a
+cast, even in the case where the representation type is nullable (even
+better: don't use a cast, call a constructor instead). Another consequence of
+the fact that the extension type is potentially non-nullable is that it
+is an error to have an instance variable whose type is an extension type,
+and then relying on implicit initialization to null.*
+
+In the body of a member of an extension type declaration _DV_ named
+`Name` and declaring the type parameters
+<code>X<sub>1</sub>, .. X<sub>s</sub></code>,
+the static type of `this` is
+<code>Name&lt;X<sub>1</sub> .. X<sub>s</sub>&gt;</code>.
+The static type of the representation name is the representation
+type.
+
+*For example, in `extension type V(R id) ...`, `id` has type
+`R`, and `this` has type `V`.*
+
+Let _DV_ be an extension type declaration named `V` with representation
+type `R`. Assuming that all types have been fully alias expanded, we say
+that _DV_ has a representation dependency on an extension type declaration
+_DV2_ if `R` contains an identifier `id` (possibly qualified) that resolves
+to _DV2_, or `id` resolves to an extension type declaration _DV3_ and _DV3_
+has a representation dependency on _DV2_.
+
+It is a compile-time error if an extension type declaration has a
+representation dependency on itself.
+
+*In other words, cycles are not allowed. This ensures that it is
+always possible to find a non-extension type which is the ultimate
+representation type of any given extension type.*
+
+The *extension type erasure* of an extension type `V` is obtained by
+recursively replacing every subterm of `V` which is an extension type by
+the corresponding representation type.
+
+*Note that this extension type erasure exists, because it is a compile-time
+error to have a dependency cycle among extension type declarations.*
+
+Let
+<code>X<sub>1</sub> extends B<sub>1</sub>, .. X<sub>s</sub> extends B<sub>s</sub></code>
+be a declaration of the type parameters of a generic entity (*it could
+be a generic class, extension type, or mixin, or typedef, or function*).
+Let <code>BB<sub>j</sub></code> be the extension type erasure of
+<code>B<sub>j</sub></code>, for _j_ in _1 .. s_.
+It is a compile-time error if
+<code>X<sub>1</sub> extends BB<sub>1</sub>, .. X<sub>s</sub> extends BB<sub>s</sub></code>
+has any compile-time errors.
+
+*For example, the extension type erasure could map
+<code>X extends C<Y>, Y extends X</code> to
+<code>X extends Y, Y extends X</code>,
+which is an error.*
+
+An extension type declaration _DV_ named `Name` may declare one or
+more constructors. A constructor which is declared in an extension type
+declaration is also known as an _extension type constructor_.
+
+*The purpose of having an extension type constructor is that it bundles
+an approach for building an instance of the representation type of an
```

Comment:
"building an instance" -> "bundling an instance", or something like this?

Reply:
Right, receiving an already fully created and initialized representation object will probably be much more common than building it in the constructor. Adjusted the text to reflect this.